src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserRoleServiceTests.cs

```diff
@@ -552,105 +933,358 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
                 Assert.Empty(userReleaseRoles);
             }
         }
-
+        
         [Fact]
-        public async Task GetAllGlobalRoles()
+        public async Task AddReleaseRole_AssignedAnalystGlobalRole()
         {
-            var role1 = new IdentityRole
+            var userId = Guid.NewGuid();
+
+            var user = new ApplicationUser
             {
-                Id = Guid.NewGuid().ToString(),
-                Name = "Role 1",
-                NormalizedName = "ROLE 1"
+                Id = userId.ToString()
             };
 
-            var role2 = new IdentityRole
+            var release = new Release
             {
-                Id = Guid.NewGuid().ToString(),
-                Name = "Role 2",
-                NormalizedName = "ROLE 2"
+                Publication = new Publication()
             };
 
             var userAndRolesDbContextId = Guid.NewGuid().ToString();
+            var contentDbContextId = Guid.NewGuid().ToString();
 
             await using (var userAndRolesDbContext = InMemoryUserAndRolesDbContext(userAndRolesDbContextId))
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
             {
-                await userAndRolesDbContext.AddRangeAsync(role1, role2);
+                await userAndRolesDbContext.AddAsync(user);
                 await userAndRolesDbContext.SaveChangesAsync();
+
+                await contentDbContext.Releases.AddAsync(release);
+                await contentDbContext.SaveChangesAsync();
             }
 
+            var emailTemplateService = new Mock<IEmailTemplateService>(Strict);
+
+            emailTemplateService.Setup(mock =>
+                    mock.SendReleaseRoleEmail(user.Email,
+                        It.Is<Release>(r => r.Id == release.Id),
+                        Contributor))
+                .Returns(Unit.Instance);
+
+            var userManager = MockUserManager();
+
+            userManager
+                .Setup(s => s.GetRolesAsync(ItIsUser(user)))
+                .ReturnsAsync(new List<string>());
+            
+            // TODO DW - should we be checking the status of this returned result?
```

Comment:
Anything you need to do still here?

Reply:
I'm going to remove the TODOs from here, as there are a bunch of other places where we interact with the UserManager service and ignore any warnings / errors in the results.  We should address these at some point, ideally altogether, but I'd say it's out of scope from this PR.

I'd raise a Jira ticket in the backlog to address this, but I think the Agile training suggests we shouldn't!